### PR TITLE
Pass context into flowgraph

### LINKED ISSUE #971: zql: invalid zng panic with cut-put-cut flowgraph
I ran into this bug while finishing up #856 (rename proc):

```
$ zq -t "head 1 | cut id | put id2=id.orig_h | cut -c id.orig_h" conn.log                                                                                                                                
#0:record[id:record[orig_h:ip,orig_p:port,resp_h:ip,resp_p:port],id2:ip]
0:[[<ZNG-ERR type ip [%!s(PANIC=String method: runtime error: slice bounds out of range [:78] with capacity 30)]: failure trying to decode IP address that is not 4 or 16 bytes long>;<ZNG-ERR type port [%!s(PANIC=String method: runtime error: slice bounds out of range [5:4])]: port encoding must be 2 bytes>;<ZNG-ERR type ip [%!s(PANIC=String method: runtime error: slice bounds out of range [6:2])]: failure trying to decode IP address that is not 4 or 16 bytes long>;]10.164.94.120;]
```

### PR DESCRIPTION
**Pull request by henridf**
closes #971

### CODE REVIEW

File: tests/suite/zeek/shared-context-bug-2.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  10.164.94.120	39681	10.47.3.155	3389	-
+
+output-format: tzng
+
+output: |
+  #0:record[id:record[orig_h:ip]]
+  0:[[10.164.94.120;]]
+
+
+
```
Comment (nwt) [inline tests/suite/zeek/shared-context-bug-2.yaml:19]:
```suggestion
```

File: tests/suite/zeek/shared-context-bug.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  #separator \x09
+  #set_separator	,
+  #empty_field	(empty)
+  #unset_field	-
+  #fields	id.orig_h	id.orig_p	id.resp_h	id.resp_p	tunnel_parents
+  #types	addr	port	addr	port	set[string]
+  10.164.94.120	39681	10.47.3.155	3389	-
+
+output-format: tzng
+
```
Comment (nwt) [inline tests/suite/zeek/shared-context-bug.yaml:13]: The default is tzng.
```suggestion
```

File: tests/suite/zeek/shared-context-bug-2.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  #separator \x09
+  #set_separator	,
+  #empty_field	(empty)
+  #unset_field	-
+  #fields	id.orig_h	id.orig_p	id.resp_h	id.resp_p	tunnel_parents
+  #types	addr	port	addr	port	set[string]
+  10.164.94.120	39681	10.47.3.155	3389	-
+
+output-format: tzng
+
```
Comment (nwt) [inline tests/suite/zeek/shared-context-bug-2.yaml:13]: The default is tzng.
```suggestion
```

File: driver/compile.go [hunk 4]
```diff
@@ -18,21 +18,24 @@ import (
...
-	return CompileWarningsChCustom(ctx, nil, program, reader, "", reverse, span, logger, ch)
+func CompileWarningsCh(ctx context.Context, zctx *resolver.Context, program ast.Proc, reader zbuf.Reader, reverse bool, span nano.Span, logger *zap.Logger, ch chan string) (*MuxOutput, error) {
+	return CompileWarningsChCustom(ctx, zctx, nil, program, reader, "", reverse, span, logger, ch)
 }
 
-func CompileWarningsChCustom(ctx context.Context, custom proc.Compiler, program ast.Proc, reader zbuf.Reader, readerSortKey string, reverse bool, span nano.Span, logger *zap.Logger, ch chan string) (*MuxOutput, error) {
+func CompileWarningsChCustom(ctx context.Context, zctx *resolver.Context, custom proc.Compiler, program ast.Proc, reader zbuf.Reader, readerSortKey string, reverse bool, span nano.Span, logger *zap.Logger, ch chan string) (*MuxOutput, error) {
+	if zctx == nil {
+		zctx = resolver.NewContext()
+	}
```
Comment (nwt) [inline driver/compile.go:38]: I think this is wrong because we always want to use the reader's context.
```suggestion
```

Comment (henridf) [reply driver/compile.go:38]: Ooops this is a leftover from when i was just trying this out in one place. Good catch!

**Pull request merged**